mp/src/game/shared/neo/weapons/weapon_neobasecombatweapon.h

```diff
@@ -211,7 +211,7 @@ class CNEOBaseCombatWeapon : public CBaseHL2MPCombatWeapon
 
 	virtual bool Deploy(void);
 
-	virtual float GetFireRate(void) OVERRIDE { Assert(false); return BaseClass::GetFireRate(); } // Should never call this base class; override in children.
+	virtual float GetFireRate() final;
```

Comment:
Singular `final` is fine/even the typical usage.

Reply:
But `final` and `override` are kind of orthogonal properties though. Consider:

```cpp
class Parent {
public:
	virtual void Foo(void);
	virtual void Bar(void);
	virtual void Baz(void);
	virtual void Foobar(void);
};

class Child : Parent {
public:
	// This is the programmer attempting to override Parent::Foo
	virtual void Foo(int param);
		// Whoops wrong signature, this will *not* override Parent::Foo(void),
		// but rather is an overload with a different signature!
		// But it's valid and won't emit a compiler error.

	// This attempts to override Parent::Bar
	virtual void Bar(int param) override;
		// Mistake caught at compile time by "override" keyword

	// Same deal as "Foo" - compiler will not catch this mistake, even if final.
	virtual void Baz(int param) final;

	// Final, and also the compiler will catch the incorrect override.
	virtual void Foobar(int param) override final;
};
```

Reply:
But I guess I'm fine with it, if you prefer this style.

Reply:
Ok I get what you mean now. Alright in this case then `override final` looks useful then. I'll change over.